java/java-impl-inspections/src/com/intellij/codeInspection/OptionalIsPresentInspection.java

```diff
@@ -131,10 +124,20 @@ private static PsiStatement extractElseStatement(@NotNull PsiIfStatement ifState
   }
 
   @Nullable
-  @Contract("null -> null")
-  static PsiReferenceExpression extractOptionalFromIsPresentCheck(PsiExpression expression) {
-    PsiMethodCallExpression call = ObjectUtils.tryCast(expression, PsiMethodCallExpression.class);
-    if (!OPTIONAL_IS_PRESENT.matches(call)) return null;
+  @Contract("null, _ -> null")
+  private static PsiReferenceExpression extractOptionalFromIsPresentCheck(PsiExpression condition, boolean[] invert) {
```

Comment:
Returning value in parameter is not a good practice in Java. Especially, using one-element array for this purpose. Is it possible to avoid such a pattern?

Reply:
It would be far more complex without it. Do you prefer an `AtomicBoolean` ?

Reply:
Of course, not. Ok, will fix it by myself.